### PR TITLE
tbb v2020.1 + update vinterface formula and vtag

### PR DESCRIPTION
**Pull request by AlexVeprev**
Checklist
* [X] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

This applies changes from https://github.com/conda-forge/tbb-feedstock/pull/59 and fixes appeared errors.

### CODE REVIEW

**Comment by AlexVeprev** [comment]
@conda-forge-admin, please rerender

**Comment by AlexVeprev** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
 
 {% set year = version.split('.')[0]|int %}
 {% set update = version.split('.')[1]|int %}
 {% if version.split('.').__len__() == 2 %}
-{%  if update == 0 %}
-{%   set vtag = "%d" % year %}
-{%  else %}
-{%   set vtag = "%d_U%d" % (year, update) %}
-{%  endif %}
+{%  set vtag = "v%d.%d" % (year, update) %}
```
Comment (anton-malakhov) [inline recipe/meta.yaml]:
```suggestion
{%  set vtag = "v%d.%d" % (year, update) %}  # since 2020
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
 {% else %}  # have to take into account for sake of the bot
 {%  set vtag = "malformed_" + version %}
 {% endif %}
-{% set vinterface = (year-2008)*1000 + update %}  # just guess, does not fit the previous versions before 2018
+{% set vinterface = (year-2009)*1000 + 100 + update %}  # just guess, does not fit the previous versions before 2020
```
Comment (anton-malakhov) [inline recipe/meta.yaml]: hm.. my guess is that the next major update might want to remove `+100` from this formula.. if there is no way to find out how `+100` appears, I'd suggest just to write the "magic major" number explicitly. I.e.:
```suggestion
{% set vinterface = 11100 + update %}  # just guess for 2020 series of updates
```

**Comment by AlexVeprev** [comment]
> Windows [test failed ](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=104734&view=logs&j=f3f71ba1-f87e-5afa-974e-577ff38d9bf3&t=76287061-6748-5183-e373-2833513eaee5&l=2138)
> Please double-check that tbb_debug.lib is packed correctly and linker can find it with cmake build process

@anton-malakhov Makefile in TBB root doesn't build debug version of TBB since 2020. Seems modification of build.py on TBB side is required in order to get debug binaries.

I see some other TBB distributions don't pack debug binaries at all (e.g. [TBB in Fedora project](https://src.fedoraproject.org/rpms/tbb/blob/master/f/tbb.spec), [TBB in Homebrew](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tbb.rb#L22)).

Do you think debug binaries are really needed in the package? If no I can modify the recipe test to use release binaries.

**Comment by anton-malakhov** [comment]
running release-only checks is fine, please do

**Comment by anton-malakhov** [comment]
We need help here @AlexVeprev .. merging for sake of other platforms meanwhile

**Pull request merged**